diff -Naur snakeyaml.orig/pom.xml snakeyaml/pom.xml
--- snakeyaml.orig/pom.xml	2011-05-27 19:26:23.000000000 +0200
+++ snakeyaml/pom.xml	2011-05-27 19:32:07.000000000 +0200
@@ -61,12 +61,12 @@
             <version>4.7</version>
             <scope>test</scope>
         </dependency>
-        <dependency>
+        <!--dependency>
             <groupId>org.springframework</groupId>
             <artifactId>spring</artifactId>
             <version>2.5.6</version>
             <scope>test</scope>
-        </dependency>
+        </dependency-->
         <dependency>
             <groupId>org.apache.velocity</groupId>
             <artifactId>velocity</artifactId>
diff -Naur snakeyaml.orig/src/test/java/examples/SpringTest.java snakeyaml/src/test/java/examples/SpringTest.java
--- snakeyaml.orig/src/test/java/examples/SpringTest.java	2011-01-12 13:56:10.000000000 +0100
+++ snakeyaml/src/test/java/examples/SpringTest.java	2011-05-27 19:34:14.000000000 +0200
@@ -18,12 +18,15 @@
 
 import junit.framework.TestCase;
 
+/* WORKAROUND FOR : springframework dependency removal
 import org.springframework.context.ApplicationContext;
 import org.springframework.context.support.ClassPathXmlApplicationContext;
+*/
 import org.yaml.snakeyaml.Yaml;
 
 public class SpringTest extends TestCase {
     public void testSimple() {
+/* WORKAROUND FOR : springframework dependency removal
         ApplicationContext context = new ClassPathXmlApplicationContext("examples/spring.xml");
         Yaml yaml = (Yaml) context.getBean("standardYaml");
         assertNotNull(yaml);
@@ -33,5 +36,6 @@
         //
         yaml = (Yaml) context.getBean("snakeYaml");
         assertNotNull(yaml);
+*/
     }
 }
\ Chybí znak konce řádku na konci souboru
diff -Naur snakeyaml.orig/src/test/java/org/yaml/snakeyaml/issues/issue9/BeanHolder.java snakeyaml/src/test/java/org/yaml/snakeyaml/issues/issue9/BeanHolder.java
--- snakeyaml.orig/src/test/java/org/yaml/snakeyaml/issues/issue9/BeanHolder.java	2011-01-12 13:56:10.000000000 +0100
+++ snakeyaml/src/test/java/org/yaml/snakeyaml/issues/issue9/BeanHolder.java	2011-05-27 19:48:24.012406769 +0200
@@ -16,7 +16,9 @@
 
 package org.yaml.snakeyaml.issues.issue9;
 
+/* WORKAROUND FOR : springframework dependency removal
 import org.springframework.core.style.ToStringCreator;
+*/
 
 public class BeanHolder {
 
@@ -41,8 +43,11 @@
 
     @Override
     public String toString() {
+/* WORKAROUND FOR : springframework dependency removal
         ToStringCreator builder = new ToStringCreator(this);
         builder.append(this.bean);
         return builder.toString();
+*/
+        return "";
     }
 }
diff -Naur snakeyaml.orig/src/test/java/org/yaml/snakeyaml/issues/issue9/Bean1.java snakeyaml/src/test/java/org/yaml/snakeyaml/issues/issue9/Bean1.java
--- snakeyaml.orig/src/test/java/org/yaml/snakeyaml/issues/issue9/Bean1.java	2011-01-12 13:56:10.000000000 +0100
+++ snakeyaml/src/test/java/org/yaml/snakeyaml/issues/issue9/Bean1.java	2011-05-27 19:48:19.260945240 +0200
@@ -16,7 +16,9 @@
 
 package org.yaml.snakeyaml.issues.issue9;
 
+/* WORKAROUND FOR : springframework dependency removal
 import org.springframework.core.style.ToStringCreator;
+*/
 
 public class Bean1 implements IBean {
 
@@ -50,10 +52,13 @@
 
     @Override
     public String toString() {
+/* WORKAROUND FOR : springframework dependency removal
         ToStringCreator builder = new ToStringCreator(this);
         builder.append(this.strVal);
         builder.append(this.intVal);
         return builder.toString();
+*/
+        return "";
     }
 
 }
diff -Naur snakeyaml.orig/src/test/java/org/yaml/snakeyaml/issues/issue9/Bean2.java snakeyaml/src/test/java/org/yaml/snakeyaml/issues/issue9/Bean2.java
--- snakeyaml.orig/src/test/java/org/yaml/snakeyaml/issues/issue9/Bean2.java	2011-01-12 13:56:10.000000000 +0100
+++ snakeyaml/src/test/java/org/yaml/snakeyaml/issues/issue9/Bean2.java	2011-05-27 19:48:49.388530893 +0200
@@ -16,7 +16,9 @@
 
 package org.yaml.snakeyaml.issues.issue9;
 
+/* WORKAROUND FOR : springframework dependency removal
 import org.springframework.core.style.ToStringCreator;
+*/
 
 public class Bean2 implements IBean {
 
@@ -46,10 +48,13 @@
 
     @Override
     public String toString() {
+/* WORKAROUND FOR : springframework dependency removal
         ToStringCreator builder = new ToStringCreator(this);
         builder.append(this.strVal);
         builder.append(this.intVal);
         return builder.toString();
+*/
+        return "";
     }
 
 }
